src/App.tsx

```diff
@@ -0,0 +1,115 @@
+import * as React from "react";
+import * as ReactDOM from "react-dom";
+import { Component, ClassAttributes } from "react";
+
+const formattedSeconds = (sec: number) =>
+  Math.floor(sec / 60) + ":" + ("0" + (sec % 60)).slice(-2);
+
+interface StopwatchProps extends ClassAttributes<Stopwatch> {
+  initialSeconds: number;
+}
+
+class Stopwatch extends Component<StopwatchProps, any> {
+  incrementer: any;
+  laps: any[];
+
+  constructor(props: StopwatchProps) {
+    super(props);
+    this.state = {
+      secondsElapsed: props.initialSeconds,
+      lastClearedIncrementer: null,
+    };
+    this.laps = [];
+  }
+
+  handleStartClick() {
+    this.incrementer = setInterval(
+      () =>
+        this.setState({
+          secondsElapsed: this.state.secondsElapsed + 1,
+        }),
+      1000
+    );
+  }
+  handleStopClick() {
+    clearInterval(this.incrementer);
+    this.setState({
+      lastClearedIncrementer: this.incrementer,
+    });
+  }
+
+  handleResetClick() {
+    clearInterval(this.incrementer);
+    (this.laps = []),
+      this.setState({
+        secondsElapsed: 0,
+      });
+  }
+
+  handleLabClick() {
```

Comment:
Typo here; it should be `Lap` instead of `Lab`.

```suggestion
  handleLapClick() {
```